recipe/meta.yaml

```
@@ -1,39 +1,67 @@
 {% set name = "emoji" %}
-{% set version = "2.2.0" %}
+{% set version = "2.12.1" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: a2986c21e4aba6b9870df40ef487a17be863cb7778dcf1c01e25917b7cd210bb
+  sha256: 4aa0488817691aa58d83764b6c209f8a27c0b3ab3f89d1b8dceca1a62e4973eb
+  patches:
+    - 0001-factor-out-testutils.patch
 
 build:
-  noarch: python
   number: 0
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
 
 requirements:
+  build:
+    - patch     # [not win]
+    - m2-patch  # [win]
   host:
-    - python >=2.7
+    - python
     - pip
+    - setuptools
```

Require bound
```suggestion
    - setuptools >=65.0.0
```

https://github.com/carpedm20/emoji/blob/v2.12.1/pyproject.toml#L2C24-L2C32

---

I suppose at this point it is a style choice whether we want to preserve all upstream requirements, even though they are not applicable to our build process. I think we go back and forth with this. We have not observed requirements for modern `setuptools` in a while.